internal/saucecloud/downloader/downloader.go

```diff
@@ -23,22 +23,31 @@ func NewArtifactDownloader(reader job.Reader, artifactConfig config.ArtifactDown
 	}
 }
 
-func (d *ArtifactDownloader) DownloadArtifact(jobID string, suiteName string, realDevice bool) []string {
-	targetDir, err := config.GetSuiteArtifactFolder(suiteName, d.config)
+func (d *ArtifactDownloader) DownloadArtifact(jobData job.Job, attemptNumber int, retries int) []string {
+	if jobData.ID == "" ||
+		jobData.TimedOut ||
+		!d.config.When.IsNow(jobData.Passed) || jobData.Status == job.StateInProgress ||
```

Comment:
Might be a bit more readable this way?
```suggestion
		jobData.TimedOut || !job.Done(jobData.Status) ||
		!d.config.When.IsNow(jobData.Passed) ||
```